AK/Traits.h

```diff
@@ -37,7 +39,7 @@ requires(IsIntegral<T>) struct Traits<T> : public GenericTraits<T> {
 };
 
 template<typename T>
-requires(IsPointer<T>) struct Traits<T> : public GenericTraits<T> {
+requires(IsPointer<T> && !IsSame<Detail::Decay<T>, char*>) struct Traits<T> : public GenericTraits<T> {
```

Comment:
Does this work for both `char*` and `char const*`? I vaguely remember decaying unfortunately doesn't actually remove the `const` from the latter:
```c++
dbgln("{}", IsSame<char*, Decay<char const*>>);
dbgln("{}", IsSame<char*, std::decay_t<char const*>>);

dbgln("{}", IsSame<char const*, Decay<char const*>>);
dbgln("{}", IsSame<char const*, std::decay_t<char const*>>);
```

Prints:
```c++
false
false
true
true
```